gtdynamics/dynamics/ChainDynamicsGraph.h

```diff
@@ -0,0 +1,113 @@
+/* ----------------------------------------------------------------------------
+ * GTDynamics Copyright 2020, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * See LICENSE for the license information
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file ChainDynamicsGraph.h
+ * @brief Builds a chain dynamics graph from a Robot object.
+ * @author Dan Barladeanu
+ */
+
+#pragma once
+
+
+//#include "gtdynamics/dynamics/OptimizerSetting.h"
+#include "gtdynamics/dynamics/DynamicsGraph.h"
+#include "gtdynamics/dynamics/Chain.h"
+#include <gtdynamics/optimizer/EqualityConstraint.h>
+#include "gtdynamics/factors/ContactDynamicsMomentFactor.h"
+#include "gtdynamics/factors/WrenchFactor.h"
+#include <gtdynamics/factors/ContactHeightFactor.h>
+
+namespace gtdynamics {
+
+using gtsam::Vector3;
+using gtsam::Vector3_;
+using gtsam::Vector6;
+using gtsam::Vector6_;
+using gtsam::Double_;
+using gtsam::Pose3;
+using gtsam::Point3;
+using gtsam::Rot3;
+using gtsam::Key;
+using gtsam::NonlinearFactorGraph ;
+
+class ChainDynamicsGraph : public DynamicsGraph {
+
+  private:
+  std::vector<std::vector<JointSharedPtr>> chain_joints_;
+  std::vector<Chain> composed_chains_;
+  double trunk_mass_;
+
+  public:
+
+    /**
+     * Constructor
+     * @param  robot      the robot
+     * @param  opt          settings for optimizer
+     * @param  angle_tolerance   tolerance for angle estimation
+     * @param  torque_tolerance   tolerance for torque estimation
+     * @param  dynamics_tolerance   tolerance for dynamics estimation
+     * @param  gravity      gravity in world frame
+     * @param  planar_axis  axis of the plane, used only for planar robot
+     */
+   ChainDynamicsGraph(
+       const Robot &robot,
+       const OptimizerSetting &opt,
+       const std::optional<Vector3> &gravity = {},
+       const std::optional<Vector3> &planar_axis = {})
+       : DynamicsGraph(opt, gravity, planar_axis),
+         chain_joints_(getChainJoints(robot)),
+         composed_chains_(getComposedChains(chain_joints_)),
+         trunk_mass_(robot.link("trunk")->mass()) {}
+
+    // Destructor
+    ~ChainDynamicsGraph() {}
+
+
+  /// Return q-level nonlinear factor graph (pose related factors)
```

Comment:
Looks like formatting is off. Do you have `clang-format` setup with GoogleStyle options? You can just run auto-format on VS Code on the whole file in that case.